src/lib/logical_query_plan/stored_table_node.cpp

```diff
@@ -93,6 +93,44 @@ bool StoredTableNode::is_column_nullable(const ColumnID column_id) const {
   return table->column_is_nullable(column_id);
 }
 
+std::shared_ptr<LQPUniqueConstraints> StoredTableNode::unique_constraints() const {
+  auto unique_constraints = std::make_shared<LQPUniqueConstraints>();
+
+  // We create unique constraints from selected table key constraints
+  const auto& table = Hyrise::get().storage_manager.get_table(table_name);
+  const auto& table_key_constraints = table->soft_key_constraints();
+
+  for (const TableKeyConstraint& table_key_constraint : table_key_constraints) {
+    // Discard constraints which involve pruned column(s)
+    const auto discard_key_constraint = [&]() {
```

Comment:
Why is this a lambda, not a simple variable?

Reply:
There is no good reason. Changed to

```cpp
// Discard key constraints that involve pruned column id(s).
const auto key_constraint_column_ids = table_key_constraint.columns();
const auto discard_key_constraint = std::any_of(_pruned_column_ids.cbegin(), _pruned_column_ids.cend(),
                                                    [&key_constraint_column_ids](const auto& pruned_column_id) {
                                                      return key_constraint_column_ids.contains(pruned_column_id);
                                                    });
```